docs/enhancements/installation-progress-bar.md

```diff
@@ -0,0 +1,164 @@
+* Will be computed in BE
+* `progres =ΣclusterStageProgress * Wi = preparingForInstallation * Wpi + installing * Wi + finalizing * Wf` while `ΣWi=1`
+  * We can manually compute an avg duration for the previous installations based on events to choose those weights and hardcode them.
+
+
+#### Preparing for installation
+
+This stage mostly generates manifests and installation configs, therefore, we will set  `preparingForInstallationProgress={1, 0}` depending on if it is done or not.
+
+
+#### Installing
+
+* This stage refers to hosts' installation
+* We will “assume” all host’s stages have a similar duration, therfore, they will be equally weighted.  
+* Hosts' progresses will be computed the same way we it is done in the current implementation, aka `hostsProgressPercent = ΣcompletedStages / ΣallStages`
+
+
+#### Finalizing
+
+* This stage refers to operators' installation
+* We will “assume” all operatos have a similar installation duration.
+* Operators' progresses will be computed the same way we it is done in the current implementation, aka `operatorsProgressPercent = completedOperators / monitoredOperators`
+
+Suggested APIs:
+* Note that those new percentage values should be reset on clusters' installation reset
+
+```
+diff --git a/swagger.yaml b/swagger.yaml
+index d53ddf59..ddbc7e9b 100644
+--- a/swagger.yaml
++++ b/swagger.yaml
+@@ -4171,6 +4171,9 @@ definitions:
+         format: date-time
+         x-go-custom-tag: gorm:"type:timestamp with time zone"
+         description: The last time that the cluster status was updated.
++      progress:
++        type: object
++        ref: '#/definitions/cluster-progress-info'
+       hosts:
+         x-go-custom-tag: gorm:"foreignkey:ClusterID;association_foreignkey:ID"
+         type: array
+@@ -4458,8 +4461,12 @@ definitions:
+   host-progress-info:
+     type: object
+     required:
++      - installation_percentage
+       - current_stage
+     properties:
++      installation_percentage:
++        type: integer
+       current_stage:
+         type: string
+         $ref: '#/definitions/host-stage'
+@@ -4477,6 +4484,21 @@ definitions:
+         x-go-custom-tag: gorm:"type:timestamp with time zone"
+         description: Time at which the current progress stage was last updated.
+
++  cluster-progress-info:
++    type: object
++    required:
++      - total_percentage
++    properties:
++      total_percentage:
++        type: integer
++      preparing_for_installation_percentage:
++        type: integer
++      installing_percentage:
++        type: integer
++      finalizing_percentage:
++        type: integer
++
+   host-stage:
+     type: string
+     enum:
+```
+
+```
+GET /api/assisted-install/v1/clusters/{cluster_id}
+{
+  "id": "3a745598-dde4-46dd-bdb1-ce7e1dea119b",
+  "progress":                                    <-------- NEW
+  {
+    "total_percentage": 21,
+    "preparing_for_installation_percentage": 100,
+    "installing_percentage": 38,
+    "finalizing_percentage": 0,
```

Comment:
Clarify that we are talking about stages here?  Otherwise installing_percentage vs. installation_percentage is confusing?

```suggestion
    "preparing_for_installation_stage_percentage": 100,
    "installing_stage_percentage": 38,
    "finalizing_stage_percentage": 0,
```

Reply:
Fixed.